CRM/Admin/Form/PaymentProcessor.php

```diff
@@ -400,4 +401,34 @@ public function updatePaymentProcessor(&$values, $domainID, $test) {
     civicrm_api3('PaymentProcessor', 'create', $params);
   }
 
+  /**
+   * Check if there is a record with the same name in the db.
+   *
+   * @param string $value
+   *   The value of the field we are checking.
+   * @param array $options
+   *   The daoName and fieldName (optional ).
+   *
+   * @return bool
+   *   true if object exists
+   */
+  public static function paymentProcessorNameExists($value, $options) {
+    $fieldName = 'name';
+    $daoName = CRM_Utils_Array::value(0, $options);
```

Comment:
@darrick 
An interesting discussing happened between my colleague ( davi ) and (Tim otten) on dev channel regarding using API in core ... it may worth to check it out :

https://chat.civicrm.org/civicrm/pl/8tisabi96pysid8j7ugo6xnrxa